Fix for DOC-450, 451, 452, 453 (#268)
Macrometacorp/docs: docs/cep/reference/extensions/execution/cache.md

---
title: Cache
---

The `cache` extension provides a persistent cache per tenant.

> ## Features

Reviewer: I recommend doing a local documentation build to see how this renders. Instructions are in the README.

The following functions are allowed:

cache:get(<key>)          - gets a value per given <key>;
cache:put(<key>, <value>) - puts <key>, <value>;
cache:count()             - counts the size of the cache;
cache:delete(<key>)       - deletes a cache for a given <key>;
cache:purge()             - invalidates/purges the current cache.

## Syntax

Cache uses the following syntax:

```js

> 	cache:put("my key", "my value");

Reviewer: Should there be spaces in the values entered?

	
	cache:get("my key")
```

> 

Reviewer: ```suggestion
```

> 

Reviewer: ```suggestion
```

## Example

```js
CREATE TRIGGER EventsPutTrigger WITH (interval=1 sec);

CREATE TRIGGER EventsGetTrigger WITH (interval=5 sec);

CREATE TABLE put_in_cache(value_is_put string);

CREATE TABLE get_from_cache(value string);

INSERT INTO put_in_cache 
SELECT cache:put("my key", "my value") as value_is_put 
FROM EventsPutTrigger;

INSERT INTO get_from_cache
SELECT cache:get("my key") as value
FROM EventsGetTrigger;
```

Following document is saved every second in `put_in_cache`.

>         {"value_is_put": "true"}

Reviewer: Should this and the value statement below be in code blocks? It kind of looks like it.

> Following document is saved every 5 seconsd in `get_from_cache`.    

Reviewer: ```suggestion
Following document is saved every five seconds in `get_from_cache`.    
```
